WalletWasabi/Wabisabi/RegistrationRequestMessage.cs

```diff
@@ -0,0 +1,68 @@
+﻿using System.Linq;
+using System.Collections.Generic;
+using NBitcoin;
+using WalletWasabi.Crypto.Groups;
+using WalletWasabi.Crypto.ZeroKnowledge;
+
+namespace WalletWasabi.Wabisabi
+{
+	/// <summary>
+	/// Represents a request message for the WabiSabi unified registration protocol.
+	/// </summary>
+	/// <remarks>
+	/// RegistrationRequestMessage message is the unified protocol message used in both phases,
+	/// inputs registration and outputs registration and it is designed to supports
+	/// credentials reissuance.
+	/// </remarks>
+	public class RegistrationRequestMessage
+	{
+		internal RegistrationRequestMessage(
+			Money balance, 
+			IEnumerable<CredentialPresentation> presented, 
+			IEnumerable<IssuanceRequest> requested, 
+			IEnumerable<Proof> proofs)
+		{
+			DeltaAmount = balance;
+			Presented = presented;
+			Requested = requested;
+			Proofs = proofs;
+		}
+
+		/// <summary>
+		/// The difference between the sum of the requested credentials and the presented credentials.
+		/// </summary>
+		/// <remarks>
+		/// A positive value of this property indicates that the request is an inputs registration request,
+		/// a negative value indicates it is a outputs registration requests while finally a zero value
+		/// indicates it is a reissuance request or well a request for zero-value credentials. 
+		/// </remarks>
+		public Money DeltaAmount { get; }
+
+		/// <summary>
+		/// Randomized credentials that will be presented for output registration or reissuance.
```

Comment:
There seem to be a discrepancy between the meaning of the comment and between the variable name `Presented`. The comment suggests that these credentials **_will be_** presented, while the name of the variable suggests that they **_have been already_** presented.